script.js

```diff
@@ -0,0 +1,155 @@
+// 1. Write the checkIfPropertyIsDefined function, which checks if the value of the property isn't undefined.
+
+const userName = {
+    firstName: 'John',
+    lastName: 'Smith',
+    age: undefined
+}
+
+function checkIfPropertyIsDefined(object, key) {
+    return object[key] !== undefined;
+}
+
+checkIfPropertyIsDefined(userName, 'firstName'); // true
+checkIfPropertyIsDefined(userName, 'age'); // false
+checkIfPropertyIsDefined(userName, 'height'); // false
+
+// 2. Add up all salaries.
+
+const salaries = {
+    john: 50,
+    adam: 100,
+    katie: 75
+}
+const salarySum = salaries.john + salaries.adam + salaries.katie;
+
+console.log(salarySum);
+
+// 3. Write the isObject function. Use the typeof operator.
+
+function isObject(object) {
+    return typeof object === 'object' && object !== null;
+}
+
+console.log(isObject({})); // true
+console.log(isObject(10)); // false
+console.log(isObject(null)); // false
+
+// 4. In the separate file.
+
+// 5. In the separate file.
+
+// 6. Write the getVoteCount function.
+
+function getVoteCount (votes) {
+    return votes.upVotes - votes.downVotes;
+}
+
+console.log(getVoteCount({ upVotes: 10, downVotes: 5 })); // 5
+console.log(getVoteCount({ upVotes: 75, downVotes: 90 })); // -15
+console.log(getVoteCount({ upVotes: 50, downVotes: 50 })); // 0
+
+// 7. Write the getCubeVolume function.
+
+function getCubeVolume(cubeDimensions) {
+    return cubeDimensions.width * cubeDimensions.length * cubeDimensions.height;
+}
+
+console.log(getCubeVolume({ width: 10, length: 5, height: 2 })); // 100
+console.log(getCubeVolume({ width: 100, length: 500, height: 0 })); // 0
+console.log(getCubeVolume({ width: 15, length: 2, height: 5 })); // 150
+
+// 8. Write the getCityInformation function, use template literals.
+
+function getCityInformation(cityInformation) {
+    return `${cityInformation.name} is in ${cityInformation.country} and has an area of ${cityInformation.areaInKilometers} square kilometers`;
+}
+
+console.log(getCityInformation({
+    name: 'Warsaw',
+    country: 'Poland',
+    areaInKilometers: 517
+})); // Warsaw is in Poland and has an area of 517 square kilometers
+
+// 9. Write the getUserCopy function. Assume that users only have the firstName and lastName.
+
+const john = {
+    firstName: 'John',
+    lastName: 'Smith'
+}
+
+function getUserCopy(user) {
+    return {
+        firstName: user.firstName,
+        lastName: user.lastName
+    };
+}
+
+const newUser = getUserCopy(john);
+console.log(newUser.firstName); // John
+console.log(newUser.lastName); // Smith
+console.log(newUser === john); // false <-- this is crucial
+
+// 10. Give you a function animal.
+// Return a string like this:
+// "This white dog has 4 legs."
+
+const animal = {
+    name: 'dog',
+    legs: 4,
+    color: 'white'
+}
+
+function getAnimalInformation(object){
```

Comment:
If we assume that the `object` is always an animal, we could create a better name for this argument